### ISSUE TITLE
Display images in comments stopped displaying images in comments

### ISSUE THREAD

**Issue by fixer1234** [soscripted/sox#132]
### Installed Version: 2.0.2DevE  Environment: Firefox/Greasemonkey 

---
###  Expected Behavior
This feature was previously working.

---
### Actual Behavior
Just noticed that the feature stopped working, but it may have started in a previous version (not a lot of comments include images).

---
### SOX Errors logged to the browser console (if available)

---
### Steps to reproduce
Here's a sample comment that displays just the link: http://superuser.com/questions/290814/purpose-of-a-mystery-key-on-an-ibm-pc-3270-keyboard#comment1638898_290814

This is on the heels of general problems with comments, apparently related to the Firefox hanging issue, so I'll try to verify that it isn't just an artifact of the other problem.

Update: I reloaded Firefox and the author status feature has been deselected.  The image is still not displayed, so it doesn't appear to be an artifact of the other problem.

**Comment by shu8** [comment]
@fixer1234 hmm the feature seems to work, it's just that it doesn't detect images in new comments/comments loaded when you click 'show more comments', which is a bug nonetheless.

I'll try getting this fixed quickly :)

**Comment by shu8** [comment]
@fixer1234 fixed in [latest dev version g](https://github.com/soscripted/sox/raw/dev/sox.user.js) :)

**Comment by fixer1234** [comment]
@shu8 Looks like it may still be a situation that if the comment is initially hidden and then comments are expanded, the image link isn't displayed.  But I did verify that if the comment is visible when the page is opened, the image is displayed.

The example I included in the issue is among a bunch of excess comments that are initially hidden.  If I click on "Show all comments" after opening the page, the image isn't rendered.  If I refresh the page, the page initially renders with those excess comments hidden, again.  Then a second or two later, all of the comments are expanded (don't know if that's the site or part of the fix you added).  However, after the comments are expanded, the image is still not shown.

**Comment by fixer1234** [comment]
@shu8 There's no reopen button on this thread, so I'll just post an update to clarify the previous post.

Going directly to the question in the thread listed earlier, the comment with the picture is hidden and you can click on Show N more comments to display the comment.  If the route to the thread is clicking on the comment link, the comment is visible when the page is rendered.  Refreshing the page returns it to the same state it was before the refresh (excess comments either displayed or hidden).  Either way, the image is not visible.  

However, if the excess comments are hidden, and the page is refreshed, and then the comments are expanded, the image is displayed.  Once the image is displayed, refreshing the page still displays the image.  But that seems to be the only way to get the image displayed.  

There is no way to re-hide excess comments, so for any route that displays the thread with the comment visible before refreshing the page, there is no way to display the image.

**Comment by shu8** [comment]
@fixer1234  I can't reproduce some of the issues you're having... here's what happens for me:

- Going direct to http://superuser.com/questions/290814/purpose-of-a-mystery-key-on-an-ibm-pc-3270-keyboard#comment1638898_290814 **doesn't** work
- Going to http://superuser.com/questions/290814/purpose-of-a-mystery-key-on-an-ibm-pc-3270-keyboard and click show more comments **does** work

---

I think you might be starting at http://superuser.com/questions/290814/purpose-of-a-mystery-key-on-an-ibm-pc-3270-keyboard and then either:

- manually adding the `#comment1638898_290814`, or 
- pasting the http://superuser.com/questions/290814/purpose-of-a-mystery-key-on-an-ibm-pc-3270-keyboard#comment1638898_290814 URL straight in on the same page

If you're doing the first thing, I think there's no point 'fixing' that, because that wouldn't happen in normal circumstances.

If you're doing the second thing, then I don't *think* there's any reason that *should* happen in normal use -- when would you paste the url of a comment when on the same page in the address bar?

If you're doing neither of those things, then it is a bug and we need to investigate more. Please confirm whether what I put before the line is (not) what you see, and also about the `#comment....` thing as well :)

**Comment by fixer1234** [comment]
@shu8 The issue doesn't refer to adding the comment or URL, as described below the line.  It's strictly about viewing the existing comment.  Right now, my system behaves like you describe in your "does" and "doesn't" bullets.  When I did the test in the previous comment, the "does" case did not behave like that.  I had to refresh the page while the comment was hidden, then click on show more comments to make the image visible.

When I posted this issue using dev E, the image wouldn't display under any condition.

BTW, another issue has also changed on its own (#134); I'll update that one.  The issue has simply disappeared after days of misbehaving with the same SOX version.  

I see that I posted this issue at dev E and did the most recent test 2 days ago.  Dev G is what is now loaded, and checking other threads, that was loaded 3 or 4 days ago.  So my test was using the same version I have loaded now, and it's behaving differently (on several features).

So it looks like there may still be a bug to fix, but we also seem to be dealing with some kind of moving target.  Is there any chance that changes they're making to SE are affecting how SOX performs?
